qlib/TableMapper.qm

```diff
@@ -682,9 +685,23 @@ on_error table_mapper.rollback();
                 map hbuf{$1.key} += $1.value, dh.pairIterator();
 
             # return nothing if nothing needs to be flushed
-            if (hbuf.firstValue().size() < insert_block)
+            *int out_list_size = getRecListSize(hbuf);
+            # note - in case hbuf by some chance contains only "constant"
+            # elements, we don't flush until the end.
+            if (!exists out_list_size || out_list_size < insert_block)
                 return;
 
+            /*
```

Comment:
sure, that's my intention, writing tests for all doubts - that's why it takes too long....